homeassistant/components/onvif/__init__.py

```diff
@@ -45,50 +45,63 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     device = ONVIFDevice(hass, entry)
 
-    try:
-        await device.async_setup()
-        if not entry.data.get(CONF_SNAPSHOT_AUTH):
-            await async_populate_snapshot_auth(hass, device, entry)
-    except (TimeoutError, aiohttp.ClientError) as err:
-        await device.device.close()
-        raise ConfigEntryNotReady(
-            f"Could not connect to camera {device.device.host}:{device.device.port}: {err}"
-        ) from err
-    except Fault as err:
-        await device.device.close()
-        if is_auth_error(err):
-            raise ConfigEntryAuthFailed(
-                f"Auth Failed: {stringify_onvif_error(err)}"
+    async with AsyncExitStack() as stack:
+        # Register cleanup callback for device
+        @stack.push_async_callback
+        async def _cleanup():
+            await _async_stop_device(hass, device)
+
+        try:
+            await device.async_setup()
+            if not entry.data.get(CONF_SNAPSHOT_AUTH):
+                await async_populate_snapshot_auth(hass, device, entry)
+        except (TimeoutError, aiohttp.ClientError) as err:
+            LOGGER.debug(
+                "Error connecting to camera %s:%s: %s",
+                device.device.host,
+                device.device.port,
+                err,
```

Comment:
Instead of adding the exception to the logger string and logging with `exc_info=True`, can you use `exc_info=err` instead? I think this will include the exception's message in the log.

Reply:
actually we don't even need this